R1.14 Bug Fix

aaronhma opened the pull request:
# https://github.com/tensorflow/tensorflow/issues/23756 
This pull request fixes documentation issues.
New URL: https://github.com/aaronhma/tensorflow-R14

This is my first pull request to Tensorflow.

MarkDaoust committed:
Apply tf1->tf2 name replaces to doc-strings and comments in tensorflow.

No code changes, only doc-strings and comments.

PiperOrigin-RevId: 244275493

MarkDaoust committed:
Apply tf1->tf2 name replaces to doc-strings and comments in tensorflow.

No code changes, only doc-strings and comments.

PiperOrigin-RevId: 244275767

kokoro-team committed:
Insert release notes place-fill

MarkDaoust committed:
Apply tf1-tf2 renames to tensorflow/python/kernel_tests docstrings and comments.

No code changes, only doc-strings and comments.

PiperOrigin-RevId: 244372113

MarkDaoust committed:
Apply tf1-tf2 renames to tensorflow/python/keras docstrings and comments.

No code changes, only doc-strings and comments.

PiperOrigin-RevId: 244379391

miaout17 committed:
Fix BatchMatMulV2 case in TFLite/Toco

PiperOrigin-RevId: 244412567

jaingaurav committed:
Parse visible device list from ConfigProto

PiperOrigin-RevId: 244476157

bananabowl committed:
First release notes pass

External contributor list remains

bananabowl committed:
Added 1.14 contributors

Command used: 

git shortlog -s 6612da89516247503f03ef76e974b51a434fb52e..f78b725d10e1386b614621465810b9e79558bd08 | cut -f 2 | sed -z 's/\n/, /g'

jaingaurav committed:
Remove blocks_per_core_limit in DeviceDescription

Querying blocks_per_core_limit requires an active CUDA context. We'd
like to avoid this requirement for DeviceDescription and keep it as
stateless as possible. Thereby allowing querying the device without
allocating any memory.

Currently blocks_per_core_limit is not being used anywhere. However, if
we'd like to add it back, we can do so with a dedicated function.

PiperOrigin-RevId: 244747460

reedwm committed:
Set mixed precision in ConfigProto if mixed precision is enabled.

Before, if a user enabled mixed precision with 'enable_mixed_precision_graph_rewrite', but explicitly passed a ConfigProto to Session, the auto_mixed_precision grappler pass would not be enabled. This change causes Session to enable auto_mixed_precision in that case.

Also, give a warning if 'enable_mixed_precision_graph_rewrite' when a Session has already been created.

PiperOrigin-RevId: 244748138

annarev committed:
Remove SparseConditionalAccumulator in 2.0.

PiperOrigin-RevId: 244751763

jaingaurav committed:
Avoid printing memory info in GetValidDeviceIds

Calling `DeviceMemoryUsage()` in `GetValidDeviceIds` is unnecessary and
leads to a dependency on the CUDA context being initialized. We'd like
to move towards having `GetValidDeviceIds` side-effect free.

PiperOrigin-RevId: 244771315

tensorflower-gardener committed:
Exclude device_tracer for CPU build

cl/243861265 introduced a dependency on tensorflow/core/platform/default/device_tracer.cc, however this file is actually empty during the CPU build because GOOGLE_CUDA is not defined. So the cc library for device_tracer is empty, and when force linking (/WHOLEARCHIVE is used for linking shared library) an empty cc library with MSVC, we hit this compiler bug:
https://support.microsoft.com/en-hk/help/4020481/fix-link-exe-crashes-with-a-fatal-lnk1000-error-when-you-use-wholearch

This is the root cause of b/130835153 and b/130748329

Since device tracer is only for GPU build, the solution is to exclude it from CPU build.

PiperOrigin-RevId: 244872492

bananabowl committed:
Update contributor list

get_external_authors -- --release=r1.14

Commit:
Update RELEASE.md

bananabowl committed:
Merge pull request #27966 from tensorflow-jenkins/create_release_notes_1.14.0

Update release notes for TensorFlow 1.14.0

kokoro-team committed:
Update version numbers to 1.14.0-rc0

bananabowl committed:
Merge pull request #27991 from miaout17/r1.14-fix-batch-matmul

Fix BatchMatMulV2 case in TFLite/Toco in 1.14 branch

bananabowl committed:
Merge pull request #28056 from annarev/cherrypicks_BKPXT

1.14.0-rc0 cherry-pick request: Remove SparseConditionalAccumulator in 2.0.

bananabowl committed:
Merge pull request #28089 from tensorflow-jenkins/create_release_notes_1.14.0

Update version numbers for TensorFlow 1.14.0-rc0

jaingaurav committed:
Remove unused StreamExecutorFactory

PiperOrigin-RevId: 244956772

jaingaurav committed:
Do not call EnablePeerAccess in GetValidDeviceIds

We call GetValidDeviceIds while listing GPU devices. This should be
side-effect free. However, currently peer access is enabled as a result
of making the call. Thus we move the call to EnablePeerAccess to be in
CreateDevices.

PiperOrigin-RevId: 244964443

bananabowl committed:
Merge pull request #28097 from reedwm/cherrypicks_O360L

1.14.0-rc0 cherry-pick request: Set mixed precision in ConfigProto if mixed precision is enabled.

tensorflower-gardener committed:
Merge pull request #28021 from avasid:master

PiperOrigin-RevId: 245066433

jaingaurav committed:
Create DeviceDescription without calling Init

We'd like to be able to get the DeviceDescription of devices like GPUs
without requiring the StreamExecutor to be initialized as this results
in GPU memory being allocated. We support this by adding a call to
CreateDeviceDescription with a given ordinal.

Note, not all platforms support this functionality.

PiperOrigin-RevId: 245090600

tensorflower-gardener committed:
Fix missing dependencies in pip package so that OSS build tests can pass. Also adds the license for the third party dependency "kissfft".

PiperOrigin-RevId: 245098837

tensorflower-gardener committed:
Merge pull request #27069 from yongtang:tf.math.floordiv-floormod

PiperOrigin-RevId: 245101673

bananabowl committed:
Pin tf-estimator-nightly to 1.14.0.dev2019042301

bananabowl committed:
Merge pull request #28120 from tensorflow/bananabowl-patch-2

Pin tf-estimator-nightly to 1.14.0.dev2019042301

bananabowl committed:
Merge pull request #28121 from annarev/cherrypicks_Z103P

1.14.0-rc0 cherry-pick request: Merge pull request #27069 from yongtang:tf.math.floordiv-floormod

bananabowl committed:
Merge pull request #28123 from bananabowl/cherrypicks_NZ9KF

Fix missing dependencies in pip package

bananabowl committed:
Merge pull request #28124 from bananabowl/cherrypicks_T3LMJ

Exclude device_tracer for CPU build

bananabowl committed:
Add a BUILD.system file for kissfft so that the license can be used as a dependency. (#28127)

PiperOrigin-RevId: 245134696

bananabowl committed:
Merge pull request #28128 from bananabowl/cherrypicks_V92W8

Parse visible device list from ConfigProto

tensorflower-gardener committed:
Remove kissfft license from the OSS licenses group as it is still causing build issues.

PiperOrigin-RevId: 245155471

allenlavoie committed:
Skip a SavedModel captured dataset loading test in v1 on gpu

It's broken not a very important use-case. Loading in v2 on GPU works.

PiperOrigin-RevId: 245172655

tensorflower-gardener committed:
Add kissfft license back and include it in the "VALID_LIBS" list in syslibs_configure.bzl.oss

PiperOrigin-RevId: 245175338

jaingaurav committed:
Avoid unnecessary GPU SteamExecutor allocation

Migrate to new DeviceDescription APIs to avoid inadvertently allocating
on unused GPUs.

PiperOrigin-RevId: 245249335

tensorflower-gardener committed:
Include kissfft 1.3.0 as a required package in setup.py.oss

PiperOrigin-RevId: 245256804

tensorflower-gardener committed:
Include the kissfft "LICENSES" folder as the "LICENSE" target.

PiperOrigin-RevId: 245261641

tensorflower-gardener committed:
Disable tests in tensorflow/examples/speech_commands for pip builds as they have a transitive dependency on kissfft (which is broken for Windows).

PiperOrigin-RevId: 245283048

bananabowl committed:
Merge pull request #28162 from reedwm/cherrypicks_T5SE0

Merge pull request #28021 from avasid:master

bananabowl committed:
Merge pull request #28159 from bananabowl/cherrypicks_ADB43

Disable tests in tensorflow/examples/speech_commands for pip builds as they have a transitive dependency on kissfft (which is broken for Windows).

yifeif committed:
Add cuda runtime 9.0 API to dlopen wrapper.

PiperOrigin-RevId: 245304515

tensorflower-gardener committed:
Undo failed attempts at including kissfft

PiperOrigin-RevId: 245315183

jaingaurav committed:
Leave GPU options untouched on CPU only setups

Despite GPU configurations being meaningless for CPU-only environments
we need to avoid any validation to maintain V1 compatibility. In V1
GPU options were simply ignored on CPU-only environments. Note, that
on GPU environments it is still correct to check for invalid indices.

PiperOrigin-RevId: 245355919

nfelt committed:
Add tf.compat.v1.all_v2_summary_ops() for TF 2.0 legacy graph users

This ports the TF 1.x tf.contrib.summary.all_summary_ops() API into the future as tf.compat.v1.all_v2_summary_ops(), and extends it to also return the new TF 2.0 summary ops that use `write()` or `write_raw_pb()`.  Including this API makes it much more convenient to use TF 2.0 summary ops along with legacy graph mode.

PiperOrigin-RevId: 245374982

tensorflower-gardener committed:
Fix find protobuf_archive

Removing -type flag because it is unnecessary and on some platform it doesn't accepted multiple types, which caused:
find: Arguments to -type should contain only one letter

Fourth try to fix #27697#issuecomment-486956391

PiperOrigin-RevId: 245433173

mdfaijul committed:
A fix to RequantizationPerChannel Op
  - Added registration for output type: quint8

bananabowl committed:
Merge pull request #28190 from bananabowl/cherrypicks_VLGMW

Three cherrypicks

jdduke committed:
Automated rollback of commit 4a464440b2e8f382f442b6e952d64a56701ab045

PiperOrigin-RevId: 245449476

jing-dong committed:
Fix memory leak in gcs file cache under certain STL versions.

The memory leak results from using vector<T>::shrink_to_fit() to attempt to
shrink the allocated space to the size used, however, the behavior of
shrink_to_fit() is implemention defined and may not actually shrink the
allocated space. This results in memory leak when Tensorflow is compiled under
certain compiler versions (e.g. g++ 5.4).

PiperOrigin-RevId: 245458291

Commit:
Avoid sorting the list of hosts by device name as _tpu_query_system_metadata now sorts devices properly by task id.

PiperOrigin-RevId: 245460584

bananabowl committed:
Use "tf-estimator-nightly" instead of "tensorflow_estimator"

bananabowl committed:
Merge pull request #28193 from tensorflow/bananabowl-patch-1

Use "tf-estimator-nightly" instead of "tensorflow_estimator"

bananabowl committed:
Merge pull request #28196 from bananabowl/cherrypicks_P69PS

Fix memory leak in gcs file cache under certain STL versions.

Commit:
[SE] Remove port::MakeUnique/WrapUnique.  Use ABSL instead.

PiperOrigin-RevId: 245834784

tensorflower-gardener committed:
Switch from TF_TENSORRT_VERSION to TF_NEED_TENSORRT environment variable to determine whether to build with TF-TRT.

Store fallback env variables for CUDA configs in bazelrc as well.

PiperOrigin-RevId: 245961133

tensorflower-gardener committed:
Merge pull request #28235 from yongtang:04282019-layers

PiperOrigin-RevId: 245961690

tensorflower-gardener committed:
Do not copy cublas header files again even if the specified path is different. This is a little awkward because we might pick up the header from the wrong (the CUDA toolkit instead of the cuBLAS) directory.

PiperOrigin-RevId: 245996105

bananabowl committed:
Merge pull request #28293 from jaingaurav/cherrypicks_gpu_memory

Cherrypicks gpu memory fixes

tensorflower-gardener committed:
Label saved_model_test as "no_pip".

PiperOrigin-RevId: 246006020

bananabowl committed:
Merge pull request #28291 from bfontain/cherrypicks_XE2UG

Avoid sorting the list of hosts by device name as _tpu_query_system_metadata now sorts devices properly by task id.

bananabowl committed:
Merge pull request #28296 from bananabowl/cherrypicks_YHIRZ

5 more cherrypicks into 1.14

gunan committed:
During cuda config detection, make find_cuda_config strip spaces around versions

PiperOrigin-RevId: 246040939

tensorflower-gardener committed:
Fix ./configure trying to find TensorRT when it shouldn't.

PiperOrigin-RevId: 246197213

bananabowl committed:
Merge pull request #28300 from bananabowl/cherrypicks_FBWYA

During cuda config detection, make find_cuda_config strip spaces around versions

bananabowl committed:
Merge pull request #28321 from bananabowl/cherrypicks_Q0L45

A fix to RequantizationPerChannel Op

MarkDaoust committed:
Apply tf1->tf2 name replaces to doc-strings and comments in tensorflow.

No code changes, only doc-strings and comments.

PiperOrigin-RevId: 246255382

bananabowl committed:
Merge pull request #28325 from bananabowl/cherrypicks_L4CBY

Two TensorRT related cherrypicks

caisq committed:
Make the --copt=-mavx and --copt=-msse4.2 flags take effect for libtensorflow builds

PiperOrigin-RevId: 246392874

bananabowl committed:
Merge pull request #28348 from bananabowl/cherrypicks_8RRCJ

Skip a SavedModel captured dataset loading test in v1 on gpu

bananabowl committed:
Merge pull request #28352 from bananabowl/cherrypicks_BBN61

Make the --copt=-mavx and --copt=-msse4.2 flags take effect for libtensorflow builds

Commit:
Fix the arguments for building libtensorflow.

PiperOrigin-RevId: 246542136

bananabowl committed:
Merge pull request #28378 from bananabowl/cherrypicks_HQ3SE

Cherrypick for doc updates and libtensorflow fix

jdduke committed:
Fix JNI interop for allowBufferHandleOutput

Fixes #28335

PiperOrigin-RevId: 246885842

tensorflower-gardener committed:
Fix FP16 build file.

PiperOrigin-RevId: 247226940

bananabowl committed:
Merge pull request #28489 from jdduke/cherrypicks_UIR7X

Fix JNI interop for allowBufferHandleOutput

bananabowl committed:
Merge pull request #28535 from reedwm/cherrypick_cuda9

1.14-rc0 cherry-pick request: Add cuda runtime 9.0 API to dlopen wrapper.

bananabowl committed:
Merge pull request #28559 from jdduke/cherrypicks_OIA1G

Fix FP16 build file.

bjacob committed:
Disable dotprod_detect on Apple.
The SIGILL-handling does not work there.

PiperOrigin-RevId: 248781016

aaroey committed:
Convert the version (which is a generator) to a list so after the check
`if not all(version):` it won't become empty.

PiperOrigin-RevId: 248791616

yifeif committed:
Do not register GPU devices if cannot dlopen CUDA/ROCm libraries.
- Only do this check when TF is built dynamically, and with GPU enabled.
- This will allow GPU machines with Nvidia driver installed but no CUDA driver to run with TensorFlow GPU pip package by falling back to CPU.

This is a feature request based on discussion in https://groups.google.com/a/tensorflow.org/d/msg/developers/MUq0ee3g-xk/EOugUvZ8AQAJ

PiperOrigin-RevId: 248816931

bjacob committed:
Restrict detect_dotprod to linux.
We're debugging further trouble on iOS even with the current
__APPLE__ check.

PiperOrigin-RevId: 248865934

bjacob committed:
Don't compile the dotprod path outside of Linux,
we don't know non-Linux devices with dotprod and we don't know
yet dotprod-runtime-detection code that runs outside of Linux.

PiperOrigin-RevId: 248866842

aaroey committed:
Automated rollback of commit 220e8e059a2514567b33cc2614fea0935bd3f631

PiperOrigin-RevId: 249052565

tensorflower-gardener committed:
Update Eigen to https://bitbucket.org/eigen/eigen/commits/a0d250e79c798a9370de97acf14fd8c780d605d7

Change TF makefile build to pull from bitbucket. Otherwise the makefile presubmit will fail until the mirror is populated, which ATM only happend after the CL is submitted.

PiperOrigin-RevId: 249071384

Commit:
Fix un-escaped character in docstring. It is causing deprecation warnings in IPython.

PiperOrigin-RevId: 249077268

Commit:
Rewrite messages of errors raised at raised graph construction to include a stack trace relative to original code before conversion.
This is done by attaching metadata (as new attributes) to errors raised in converted code. This metadata includes the necessary information to indicate the location in the original code that corresponds to these exceptions' actual stack trace. The location is then included in the error message.
In the common cases, the type of the error raised is preserved, however unknown error types will be replaced by a dedicated AutoGraph error when leaving the converted code. The error type remains the same within the converted code.

The approach is much less intrusive than wrapping an error, as it allows preserving the exception type within graph code. The risk of conflicting with existing attributes is extremely low, and since the names are internal, they are easily changed.
However, it would be better to reserve standard names for this.

PiperOrigin-RevId: 249109716

joker-eph committed:
[TF:XLA] Add XLA lowering for SelectV2

PiperOrigin-RevId: 249112081

bananabowl committed:
Pull up 1.14 to a071e8434eb695e61e4f70e8c001e775387c5e2a 'git merge --strategy-option theirs a071e8434eb695e61e4f70e8c001e775387c5e2a'

bananabowl committed:
remove dupe no_pip tag

bananabowl committed:
Merge pull request #28885 from tensorflow/r1.14-pull-3

 Pull 1.14 up to a071e8434eb695e61e4f70e8c001e775387c5e2a

bananabowl committed:
Merge pull request #28887 from mdanatg/cherrypicks_16GG8

Cherrypicks: error rewriting and fix for deprecation warning

bananabowl committed:
Merge pull request #28889 from bananabowl/cherrypicks_7YB79

3 cherrypicks for 1.14

yifeif committed:
Add missing -DGOOGLE_CUDA to dsoloader.

PiperOrigin-RevId: 249238691

Commit:
Fix assert_shallow_structure length checks for non map collections.

This prevents issues where data could be silently dropped during flatten operations:

shallow_tree = [(None,), (None,)]
input_tree = [(1,), (2,), (3,)]
nest.flatten_up_to(shallow_tree, input_tree) -> [(1,), (2,)]

Added optional flag check_full_subtree which can be disabled to allow nodes in the tree to be treated as key-value maps to allow previous behavior.

PiperOrigin-RevId: 249261381

mihaimaruseac committed:
Update png_archive version to 1.6.37

PiperOrigin-RevId: 249272809

bananabowl committed:
Merge pull request #28905 from tensorflow/mm-png-version-update

Update png_archive version to 1.6.37

alextp committed:
Fix gradient of where_v2 to not invoke one-argument where which is not XLA-friendly.

PiperOrigin-RevId: 249279381

bananabowl committed:
Internal change

PiperOrigin-RevId: 249283936

josh11b committed:
Set the forward-compat date to the future.

This way we pick up all changes that were only being delayed for
forward-compatibility reasons.

bananabowl committed:
Merge pull request #28908 from bananabowl/cherrypicks_VVM3J

3 cherrrypicks

bananabowl committed:
Merge pull request #28906 from tensorflow/josh11b-patch-1

Set the forward-compat date to the future.

qlzh727 committed:
Fix LSTM state issue for 2.7.

PiperOrigin-RevId: 249310963

bananabowl committed:
Internal change

PiperOrigin-RevId: 249313553

bananabowl committed:
Merge pull request #28913 from bananabowl/cherrypicks_BN094

Remove data dependencies so that these two tests can be built

tanzhenyu committed:
Move the optimizer name scope from model.training to optimizer

PiperOrigin-RevId: 249328139

cheshire committed:
Fix a segfault in StreamExecutorMemoryAllocator in a multi-GPU case

PiperOrigin-RevId: 249335436

bananabowl committed:
Merge pull request #28914 from bananabowl/cherrypicks_WTDPO

Move the optimizer name scope from model.training to optimizer

tensorflower-gardener committed:
Fix Windows GPU build

Rename cublas_virtual_headers to cublas_headers

Fix linking error by exporting commonly used absel symbols

PiperOrigin-RevId: 249426736

bananabowl committed:
Merge pull request #28927 from bananabowl/cherrypicks_5PP1F

Two more fixes

facaiy committed:
BUG: tf.function cannot handle exception raised by custom op

Commit:
Add ClusterResolver symbols to the TF 2 upgrade tool.

PiperOrigin-RevId: 249487525

tensorflower-gardener committed:
Removed the test for "tf.gather does not allow indices with unknown rank when batch_dims is specified."

PiperOrigin-RevId: 249489511

bananabowl committed:
Merge pull request #28943 from bananabowl/cherrypicks_3BI4C

Two more cherrypicks

er91 committed:
Rename TPUReplicate op to _TPUReplicate so it won't be accidentally serialized.

Renaming this op is fine because it's only a transient op occurring in GraphOptimizationPass and it's never serialized in GraphDef.

PiperOrigin-RevId: 249517076

guptapriya committed:
Make `trainable` argument of variables default to `None` everywhere consistently, and update the documentation to be consistent as well.
Currently, in practice it is `None` (due to the way VariableMetaclass is used), but in some APIs, it is listed as `True` which can be a source of confusion and appears in the generated documentation. In practice, this should have no effect other than fixing documentation.
Added a test to illustrate the behavior, but the test passes with and without the changes here.

PiperOrigin-RevId: 249559719

aaronhma commented:
I signed it!

facaiy committed:
BLD: add build file

tensorflower-gardener committed:
Added support for freezing V1 reference variable to constant node, and satisfy grappler by removing the initializer_name of the reference variables.

PiperOrigin-RevId: 249580046

facaiy committed:
TST: clean test case

bananabowl committed:
Merge pull request #28960 from yifeif/cherrypicks_WGLB7

Release r1.14rc0 cherry-pick request: make tensorflow-gpu work on machine w/o CUDA setup

caisq commented:
@aaronhma Your change list doesn't match your PR description. The description seems to indicate this is a simple documentation change, but your PR actually includes many commits and touches many files. Can you check whether this is correct?

bananabowl committed:
Merge pull request #28965 from bananabowl/cherrypicks_XFZ0N

3 more cherrypicks

bananabowl committed:
Merge pull request #28970 from bananabowl/cherrypicks_SSO7Z

BUG: tf.function cannot handle exception raised by custom op

bananabowl committed:
Internal change

PiperOrigin-RevId: 249664015

bananabowl committed:
Merge pull request #28974 from jdduke/cherrypicks_L1BTW

Cherry-pick for fixes for dotprod detection on iOS

bananabowl committed:
Merge pull request #28973 from bananabowl/cherrypicks_NL20I

Two cherrypicks

bananabowl committed:
Re-add 1.14 release notes

Somehow this got lost in the fast forward. This is incomplete and does not mention changes from the last few weeks as it was created before the fast forward. Will update it soon.

bananabowl committed:
Merge pull request #28986 from tensorflow/bananabowl-patch-1

Re-add 1.14 release notes

bananabowl committed:
Remove encoding kwarg passed to bytes

This is only available in python 3

bananabowl committed:
Merge pull request #29000 from tensorflow/bananabowl-patch-1

Remove encoding kwarg passed to bytes

rthadur commented:
@aaronhma please check caisq@ comments and open a new PR, as this PR includes lot of commits , thanks for your contribution.
